### PR TITLE
Redesign custom filter rows

### PR DESCRIPTION
**Pull request by TomTirapani**
+ Use multi-select for multi-value operators
+ Remove confusing implicit join message

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by lbwexler** [comment]
Love the direction this is moving -- think the multi-select much clearer than the current strategy.

But having seen it, not 100% sure about the new 2 line layout here  (which I pushed us to try).  Especially for the number filtering, it seems a little sub-optimal and confusing.  Wonder what the growing multi-select would look like within the three column layout.

As an orthogonal matter, I would remove the clear icon within the inputs -- it seems confusing and redundant, given the ability to just zap the entire filter entry with the red minus.

**Comment by lbwexler** [comment]
Actually for the text columns,  the current layout is growing on me.  I think if you used the words `equals` and `not equals` now that you have the space, it would work really well.

A little less sure about what to do with numbers -- we definitely lost the ability to have multiple values for = and != -- not sure we thought  about that.  Putting that aside, as long as numbers only allow single values, could imagine going back to the single line layout.

Sorry for the ramble....

**Comment by TomTirapani** [comment]
Good point about losing the ability to have multiple `=` and `!=` values for numeric fields, the same would apply to date fields :|

In light of that, not sure if this multi-select approach is going to work. For those types of values we'd need to allow multiple rows, which would bring us back to square one in terms of AND|OR confusion. I suppose we could use multi-select for those types of values too (at the cost of number formatting / date picker), and make each entry on the custom tab similar to a FilterChooser (could even leverage the same `valueParser` / `valueRenderer` configs on the FieldSpec), but I don't think the loss of functionality is justified.

If we do stick with the multi-select approach, I would vote for the same UI (2 line or inline) across all the types. I think switching across types would be weird, but could be persuaded.

Should we sit down with Anselm and discuss the options we have and weight up their pros and cons?

**Comment by lbwexler** [comment]
This all makes sense -- yes, let's sit with Anselm and make some decisions
here.

-Lee

**Comment by lbwexler** [comment]
Looks prmising -- could see using a less wonky tag for the non displayable filter's in FilterChooser  -- e.g. "Custom Filter on TradeDate"

**Comment by lbwexler** [comment]
Would like to review in detail in person

File: cmp/grid/filter/GridFilterModel.js [hunk 1]
```diff
@@ -175,4 +183,18 @@ export class GridFilterModel extends HoistModel {
             });
         });
     }
+
+    getOuterCompoundFilter(filter, field) {
+        if (!filter?.isCompoundFilter) return null;
+
+        // This is the outer compound filter if all its children
+        // are FieldFilters on the matching field.
+        if (every(filter.filters, it => it.field === field)) {
```
Comment (lbwexler) [inline cmp/grid/filter/GridFilterModel.js:192]: second clause to every can just be [field]

**Review by lbwexler — APPROVED** [review]

**Comment by lbwexler** [comment]
Happy to go forward with this -- its a minimal change for the user experience, a nice functional enhancement, and a nice cleanup of our underlying model.  @amcclain  -- let us know if you want to take a look

**Pull request merged**